### ISSUE TITLE
tmp folder size grow by 'send to nextcloud' upload

### ISSUE THREAD

**Issue by mr-bolle** [nextcloud/android#4341]
### Steps to reproduce
1. open a Filemanager, Galerie and select files what you want to upload
2. select share with - Nextcloud
3. after upload complete upload check the temp folder   [ex. storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/tmp/USER@NEXTCLOUD.DOMAIN.com]
4. a full copy of the uploaded files still exist

**Note:** i don't use a external SD Card

### Expected behaviour
If the Upload are complete the files into the tmp folder sould be removed

### Actual behaviour
After few month the size of this folder grow up to 6-15 GB

## Server configuration detail

**Operating system:** Linux 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u4 (2019-07-19) x86_64

**Webserver:** Apache/2.4.25 (Debian) (apache2handler)

**Database:** mysql 10.4.7

**PHP version:** 

7.3.7
Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, apcu, exif, gd, imagick, intl, ldap, memcached, pcntl, pdo_mysql, pdo_pgsql, redis, sodium, zip, Zend OPcache

**Nextcloud version:** 15.0.10 - 15.0.10.0

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:** https://github.com/nextcloud/docker/

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.1.0
 - activity: 2.8.2
 - admin_audit: 1.5.0
 - bookmarks: 1.1.1
 - bruteforcesettings: 1.4.0
 - calendar: 1.6.5
 - cloud_federation_api: 0.1.0
 - comments: 1.5.0
 - contacts: 3.1.3
 - dav: 1.8.2
 - deck: 0.6.6
 - drawio: 0.9.3
 - federatedfilesharing: 1.5.0
 - federation: 1.5.0
 - files: 1.10.0
 - files_external: 1.6.0
 - files_markdown: 2.0.6
 - files_pdfviewer: 1.4.0
 - files_sharing: 1.7.0
 - files_texteditor: 2.7.0
 - files_trashbin: 1.5.0
 - files_versions: 1.8.0
 - files_videoplayer: 1.4.0
 - firstrunwizard: 2.4.0
 - gallery: 18.2.0
 - issuetemplate: 0.5.0
 - keeweb: 0.5.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.3.0
 - nextcloud_announcements: 1.4.0
 - notifications: 2.3.0
 - oauth2: 1.3.0
 - password_policy: 1.5.0
 - polls: 0.10.2
 - provisioning_api: 1.5.0
 - ransomware_protection: 1.3.0
 - richdocuments: 3.3.15
 - sharebymail: 1.5.0
 - spreed: 5.0.4
 - support: 1.0.0
 - survey_client: 1.3.0
 - systemtags: 1.5.0
 - tasks: 0.11.1
 - theming: 1.6.0
 - twofactor_backupcodes: 1.4.1
 - twofactor_totp: 2.1.2
 - updatenotification: 1.5.0
 - workflowengine: 1.5.0
Disabled:
 - encryption
 - serverinfo
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "htaccess.RewriteBase": "\/",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "apps_paths": [
        {
            "path": "\/var\/www\/html\/apps",
            "url": "\/apps",
            "writable": true
        },
        {
            "path": "\/var\/www\/html\/custom_apps",
            "url": "\/custom_apps",
            "writable": true
        }
    ],
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "domain1.nextcloud.com",
        "domain2.nextcloud.com"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/domain1.nextcloud.com",
    "dbtype": "mysql",
    "version": "15.0.10.0",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "theme": "",
    "loglevel": 1,
    "maintenance": false,
    "mail_smtpmode": "smtp",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpsecure": "tls",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "trashbin_retention_obligation": "30, 90",
    "versions_retention_obligation": "30, 180"
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** 

**Operating system:** 

## Logs

<details><summary>Web server error log</summary>

```
no Webserver error log
```
</details>

<details><summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

<details><summary>Browser log</summary>

Insert your browser log here, this could for example include:

	a) The javascript console log
	b) The network log
	c) ...
</details>

**Comment by tobiasKaminsky** [comment]
Thanks for your bug report.
This will be fixed with next version (3.8.0), but you can test it already with https://github.com/nextcloud/android/pull/4349
There will be soon a APK file you can install in parallel.

**Comment by mr-bolle** [comment]
Thanks @tobiasKaminsky your fix helps

**my tests**
1. Third App Upload
2. Upload from Nextcloud App
3. *Auto upload* with Option *keep in original folder*
4. *Auto upload* with Option *delete*

All tests successfully done - the temp folder are empty after successful upload

**Comment by mr-bolle** [comment]
fixed #4341

**Comment by mr-bolle** [comment]
Hi, i found a other issue with the nextcloud qa app (version 10456) [apk nextcloud qa](https://github.com/nextcloud/android/pull/4349#issuecomment-522684043)

If i download a file from the nextcloud app i have follow folder structure

*/storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/tmp* temp upload folder
*/storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/USER@NEXTCLOUD.DOMAIN* Path for downloadet content 

with the nextcloud qa app (version 10456) and if i download files from the nextcloud, any files are stored into the tmp folder
*/storage/emulated/0/Android/media/com.nextcloud.android.qa/nextcloud/tmp* if i use multi accounts, any files should be stored in one folder.

**Comment by tobiasKaminsky** [comment]
Thanks for testing :+1:

**Comment by tobiasKaminsky** [comment]
@mr-bolle as this is unrelated, please open up a new issue next time.
It does not hurt to have many different issues, but it helps a lot to keep track on all issues ;-)
Please see #4352 for further discussions
